queue_services/entity-filer/src/entity_filer/resources/worker.py

```diff
@@ -432,7 +439,11 @@ def process_filing(
         #     )
 
     if any("registration" in x for x in legal_filings):
-        filing_submission.legal_entity_id = business.id
+        if alternate_name.entity_type == BusinessCommon.EntityTypes.SOLE_PROP:
```

Comment:
Is it possible for `alternate_name` to be `None` here? If so you may need to also check it exists before getting the entity type.